sympy/physics/control/lti.py

```diff
@@ -3217,3 +3217,444 @@ def expand(self, **hints):
+    B : Matrix
+        The Input-to-State matrix of the state space model.
+    C : Matrix
+        The State-to-Output matrix of the state space model.
+    D : Matrix
+        The Feedthrough matrix of the state space model.
+
+    Examples
+    ========
+
+    >>> from sympy import Matrix
+    >>> from sympy.physics.control import StateSpace
+
+    The easiest way to create a StateSpaceModel is via four matrices:
+
+    >>> A = Matrix([[1, 2], [1, 0]])
+    >>> B = Matrix([1, 1])
+    >>> C = Matrix([[0, 1]])
+    >>> D = Matrix([0])
+    >>> StateSpace(A, B, C, D)
+    StateSpace(Matrix([
+    [1, 2],
+    [1, 0]]), Matrix([
+    [1],
+    [1]]), Matrix([[0, 1]]), Matrix([[0]]))
+
+
+    One can use less matrices. The rest will be filled with a minimum of zeros:
+
+    >>> StateSpace(A, B)
+    StateSpace(Matrix([
+    [1, 2],
+    [1, 0]]), Matrix([
+    [1],
+    [1]]), Matrix([[0, 0]]), Matrix([[0]]))
+
+
+    See Also
+    ========
+
+    TransferFunction, TransferFunctionMatrix
+
+    References
+    ==========
+    .. [1] https://en.wikipedia.org/wiki/State-space_representation
+    .. [2] https://in.mathworks.com/help/control/ref/ss.html
+
+    """
+    def __new__(cls, A=None, B=None, C=None, D=None):
+        if(A == None and B == None and C == None and D == None):
+            A = B = C = D = zeros(1)
+        elif(B == None and C == None and D == None):
+            B = zeros(A.rows,1)
+            C = zeros(1, A.cols)
+            D = zeros(1)
+        elif(C == None and D == None):
+            C = zeros(1, A.cols)
+            D = zeros(1, B.cols)
+        elif(D == None):
+            D = zeros(C.rows, B.cols)
+
+        A = _sympify(A)
+        B = _sympify(B)
+        C = _sympify(C)
+        D = _sympify(D)
+
+        if (isinstance(A, ImmutableMatrix) and isinstance(B, ImmutableDenseMatrix) and
+            isinstance(C, ImmutableDenseMatrix) and isinstance(D, ImmutableDenseMatrix)):
+            # Check State Matrix is square
+            if A.rows != A.cols:
+                raise ShapeError("Matrix A must be a square matrix.")
+
+            # Check State and Input matrices have same rows
+            if A.rows != B.rows:
+                raise ShapeError("Matrices A and B must have the same number of rows.")
+
+            # Check Ouput and Feedthrough matrices have same rows
+            if C.rows != D.rows:
+                raise ShapeError("Matrices C and D must have the same number of rows.")
+
+            # Check State and Ouput matrices have same columns
+            if A.cols != C.cols:
+                raise ShapeError("Matrices A and C must have the same number of columns.")
+
+            # Check Input and Feedthrough matrices have same columns
+            if B.cols != D.cols:
+                raise ShapeError("Matrices B and D must have the same number of columns.")
+
+            obj = super(StateSpace, cls).__new__(cls, A, B, C, D)
+            obj._A = A
```

Comment:
You shouldn't set the properties of the objects.
This can be derived from `self.args`

Reply:
Yes, It can be derived from `self.args` but doing it explicitly is followed in `lti.py`.I don't see any issues with that.
For example the `TransferFunction` class - 
https://github.com/sympy/sympy/blob/68f7e71c5c22948c1423589ba179c5080a596768/sympy/physics/control/lti.py#L462-L465

Reply:
I think that we need to refactor change to TransferFunction to follow that design as well.

Reply:
To address this as discussed in our latest meeting, I would stick to setting the variables explicitly only. 
I found multiple comments in previous PRs related to the control module. Here is one of them asking to change the initial implementation with args  - https://github.com/sympy/sympy/pull/18436#discussion_r426194974 .I remember taking it into consideration when I designed the class. All the classes in `lti.py` are consistent with this.